### PR TITLE
added the envoyLogLevel field, along with it's tests

### PR DESCRIPTION
**Pull request by irateswami**
# Description

The `gloo.gatewayProxies.gatewayProxy.logLevel` field in helm did not do anything. To set the gateway proxy log level, the user must use the `envoyLogLevel` field. This commit adds that field and it's corresponding tests, which are passing. 

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: changelog/v1.14.0-beta7/change-loglevel-to-envoyloglevel.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+changelog:
+  - type: HELM
+    description: >
+       Added `gloo.gatewayProxies.gatewayProxy.envoyLogLevel` to the helm values and deprecated `gloo.gatewayProxies.gatewayProxy.logLevel` as the latter has no use in setting the gateway-proxy log level.
```
Comment (sam-heilbron) [inline changelog/v1.14.0-beta7/change-loglevel-to-envoyloglevel.yaml:4]: I think we should refer to this just as `gatewayProxies.gatewayProxy.envoyLogLevel`. The `gloo` prefix is used when referencing a value in a Helm subchart, which is what an enterprise user would do, but this is the open source repo so I think we're good to just have the values match what open source users would configure

File: docs/content/reference/values.txt [hunk 2]
```diff
@@ -794,7 +795,8 @@
 |gatewayProxies.gatewayProxy.istioMetaMeshId|string||ISTIO_META_MESH_ID Environment Variable. Defaults to "cluster.local"|
 |gatewayProxies.gatewayProxy.istioMetaClusterId|string||ISTIO_META_CLUSTER_ID Environment Variable. Defaults to "Kubernetes"|
 |gatewayProxies.gatewayProxy.istioDiscoveryAddress|string||discoveryAddress field of the PROXY_CONFIG environment variable. Defaults to "istiod.istio-system.svc:15012"|
-|gatewayProxies.gatewayProxy.logLevel|string||Level at which the pod should log. Options include "info", "debug", "warn", "error", "panic" and "fatal". Default level is info|
+|gatewayProxies.gatewayProxy.logLevel|string||DEPRECATED, use envoyLogLevel instead. Level at which the pod should log. Options include "info", "debug", "warn", "error", "panic" and "fatal". Default level is info|
```
Comment (sam-heilbron) [inline docs/content/reference/values.txt:798]: I would recommend trying out what happens if we delete this field, and seeing how it impacts users. If we claim that this field, if configured, had no effect on a Gloo installation, let's verify the behavior if we just delete the field altogether. For the API of our Gloo Edge resources (Gateway, VirtualService...etc) there are some stricter requirements, but I think for our Helm API we may be safe to delete the field

File: install/helm/gloo/generate/values.go [hunk 3]
```diff
@@ -423,7 +423,8 @@ type GatewayProxy struct {
 	IstioMetaMeshId                *string                      `json:"istioMetaMeshId,omitempty" desc:"ISTIO_META_MESH_ID Environment Variable. Defaults to \"cluster.local\""`
 	IstioMetaClusterId             *string                      `json:"istioMetaClusterId,omitempty" desc:"ISTIO_META_CLUSTER_ID Environment Variable. Defaults to \"Kubernetes\""`
 	IstioDiscoveryAddress          *string                      `json:"istioDiscoveryAddress,omitempty" desc:"discoveryAddress field of the PROXY_CONFIG environment variable. Defaults to \"istiod.istio-system.svc:15012\""`
-	LogLevel                       *string                      `json:"logLevel,omitempty" desc:"Level at which the pod should log. Options include \"info\", \"debug\", \"warn\", \"error\", \"panic\" and \"fatal\". Default level is info"`
+	LogLevel                       *string                      `json:"logLevel,omitempty" desc:"DEPRECATED, use envoyLogLevel instead. Level at which the pod should log. Options include \"info\", \"debug\", \"warn\", \"error\", \"panic\" and \"fatal\". Default level is info"`
```
Comment (sam-heilbron) [inline install/helm/gloo/generate/values.go:426]: Where in the project is this value consumed or do we not have any helm templates which tried to parse this value?

**Review by sam-heilbron — COMMENTED** [review]
Looking good! I think the unit tests are a great way to validate that a user defined value gets injected into the helm template properly. However, I think the area that we're still missing for a test (even a manual one), is demostrating that the user defined value actually changes the log level on the running instance. Can we either include some manual test in the PR outline to demonstrate what steps a user could take to validate this, or an automated test that perhaps looks for a certain log line that will only be available with debug logging enabled?

File: docs/content/reference/values.txt (hunk 2 above)
Comment (nfuden) [reply docs/content/reference/values.txt:798]: If it did nothing and does not break anything if it is specified even after removal. Then there is no reason to leave the tombstone imo

File: changelog/v1.14.0-beta7/change-loglevel-to-envoyloglevel.yaml (hunk 1 above)
Comment (nfuden) [reply changelog/v1.14.0-beta7/change-loglevel-to-envoyloglevel.yaml:4]: Should this have an issuelink?

File: install/test/helm_test.go [hunk 4]
```diff
@@ -4732,7 +4790,10 @@ metadata:
 					})
 
 					It("has a creates a deployment", func() {
-						prepareMakefile(namespace, helmValues{})
+						prepareMakefile(namespace, helmValues{
+							valuesArgs: []string{
+								"gatewayProxies.gatewayProxy.envoyLogLevel=debug",
```
Comment (nfuden) [inline install/test/helm_test.go:4795]: What are we testing with this extra value?

File: install/helm/gloo/generate/values.go [hunk 5]
```diff
@@ -423,7 +423,8 @@ type GatewayProxy struct {
...
+	EnvoyLogLevel                  *string                      `json:"envoyLogLevel,omitempty" desc:"Level at which the pod should log. Options include \"info\", \"debug\", \"warn\", \"error\", \"panic\" and \"fatal\". Default level is info"`
```
Comment (nfuden) [inline install/helm/gloo/generate/values.go:427]: These dont look like the actual supported log levels in envoy https://www.envoyproxy.io/docs/envoy/latest/start/quick-start/run-envoy

File: install/helm/gloo/generate/values.go (hunk 3 above)
Comment (sam-heilbron) [reply install/helm/gloo/generate/values.go:426]: Given that it's consumed in the template to set an env variable, can we confirm if setting the env variable has any effect (we can compare the bheavior with and without it set), and if it does nothing, I think we should just delete it

File: install/test/helm_test.go (hunk 4 above)
Comment (nfuden) [reply install/test/helm_test.go:4773]: Ok so what we are saying is then is the test changes that were added are just no ops that help with dependencies or something?
I just dont quite understand but if we all agree the tests arent meant to be doing things then fine

File: install/helm/gloo/generate/values.go (hunk 3 above)
Comment (nfuden) [reply install/helm/gloo/generate/values.go]: I am confused this conversation is resolved and it says we are removing it. However I see the field still. Is this intentional?

File: install/test/helm_test.go [hunk 6]
```diff
@@ -4784,17 +4823,6 @@ metadata:
 
 					})
 
-					It("can set log level env var", func() {
```
Comment (sam-heilbron) [inline install/test/helm_test.go:4787]: Why are we deleting this test? This test seems to test that we set the log level of the discovery deployment (via the LOG_LEVEL env variable) appropriately, which we still do. Your changes affect a different component/deployment, so I would not expect this test to be affect. Am I missing something?

File: install/test/helm_test.go [hunk 7]
```diff
@@ -4554,17 +4601,6 @@ metadata:
 						})
 
 					})
-
-					It("can set log level env var", func() {
```
Comment (sam-heilbron) [inline install/test/helm_test.go:4558]: Same as below, I don't think this deployment should be affected

File: install/test/helm_test.go [hunk 8]
```diff
@@ -3515,11 +3551,11 @@ spec:
 					It("can set log level env var", func() {
 						gatewayProxyDeployment.Spec.Template.Spec.Containers[0].Env = append(
 							gatewayProxyDeployment.Spec.Template.Spec.Containers[0].Env,
-							GetLogLevelEnvVar(),
```
Comment (sam-heilbron) [inline install/test/helm_test.go:3518]: Given that we're deleting this, should we be changing the name of this test? This test seems to test that we can set the log level to a new value, which now is independent of env var, so seems like "can set log level env var" is no longer accurate

File: install/test/helm_test.go [hunk 9]
```diff
@@ -2661,7 +2653,7 @@ spec:
 						}
 						container := GetQuayContainerSpec("gloo-envoy-wrapper", version, GetPodNamespaceEnvVar(), podname)
 						container.Name = "gateway-proxy"
-						container.Args = []string{"--disable-hot-restart"}
+						container.Args = []string{"--disable-hot-restart", "--log-level debug"}
```
Comment (sam-heilbron) [inline install/test/helm_test.go:2656]: This seems to set the log level to debug always. Should we keep it as info (of just non-existent as the default behavior is) and then we can assert that when we supply a helm value, the container arg is set?

**Review by sam-heilbron — COMMENTED** [review]
Couple small aspects about the tests to validate this change, but overall changes look good!

File: changelog/v1.14.0-beta7/change-loglevel-to-envoyloglevel.yaml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+changelog:
+  - type: HELM
+    description: >
+       Added `gatewayProxies.gatewayProxy.envoyLogLevel` to the helm values and deprecated `gloo.gatewayProxies.gatewayProxy.logLevel` as the latter has no use in setting the gateway-proxy log level.
```
Comment (jenshu) [inline changelog/v1.14.0-beta7/change-loglevel-to-envoyloglevel.yaml:4]:
```suggestion
       Added `gatewayProxies.gatewayProxy.envoyLogLevel` to the helm values and removed `gloo.gatewayProxies.gatewayProxy.logLevel` as the latter has no use in setting the gateway-proxy log level.
```
```suggestion
       Added `gatewayProxies.gatewayProxy.envoyLogLevel` to the helm values and deprecated `gloo.gatewayProxies.gatewayProxy.logLevel` as the latter has no use in setting the gateway-proxy log level.
```

File: install/test/helm_test.go [hunk 11]
```diff
@@ -3513,12 +3511,12 @@ spec:
 					})
 
 					It("can set log level env var", func() {
```
Comment (jenshu) [inline install/test/helm_test.go:3513]:
```suggestion
					It("can set envoy log level", func() {
```

**Review by jenshu — APPROVED** [review]

**Review by jenshu — APPROVED** [review]

**Review by ianmacclancy — APPROVED** [review]
Looks good!

**Comment by nfuden** [comment]
What version do we intend to backport this to? It feels like a bug and therefore a backport to 1.11 prior to having resolve issue set

**Pull request merged**